### ISSUE TITLE
Is it possible to use the P2P dataset to train LiverMatch?

### ISSUE THREAD

**Issue by AlanConnorChen** [zixinyang9109/LiverMatch#12]
Hello, I configured the LiverMatch environment according to the instructions in the README, changed the config path to `configs/new_task3_004_002.yaml`. During training, the dimensions of the predictions and the ground truth do not match, as shown in the figure.

<img width="825" height="554" alt="Image" src="https://github.com/user-attachments/assets/270d0994-4170-44f5-8ba5-d8c0780b9f61" />

As a result, the following error occurs:
```
Using a target size (torch.Size([5065])) that is different to the input size (torch.Size([3276])) is deprecated. Please ensure they have the same size.
TRAIN jump
```
The following code in `liver_trainer.py` catches it:
```python
except Exception as inst:
    print(inst)
    print("TRAIN jump")
```
So, can the dataset in P2P be used to train LiverMatch? According to the P2P paper, I believe this dataset can be used for training. Is there something misconfigured?

**Comment by zixinyang9109** [comment]
The answer is sure! Something should be misconfigured.

It should be very easily reproduced unless I forgot to update something in the github version.

Could you please load an examlpe via the datasets/liver_task3.py and check their dimensions?

**Comment by zixinyang9109** [comment]
The loss is used for visibility scores. You can turn it off for now. 

The datasets/liver_task3.py  searches correspondences and the following code set the label:  
       # only src scores
        src_idx = list(set(correspondence[:, 0].int().tolist()))
        src_gt = torch.zeros(src_pcd.size(0))
        src_gt[src_idx] = 1.
        src_gt_labels = src_gt.to(torch.device('cuda'))
        vis_loss, vis_cls_precision, vis_cls_recall = self.get_weighted_bce_loss(scores_vis, src_gt_labels)
        loss_info.update({'vis_loss': vis_loss, 'vis_recall': vis_cls_recall, 'vis_precision': vis_cls_precision})

**Comment by zixinyang9109** [comment]
If the datasets/liver_task3.py searches the correspondences based on the point clouds used for the network, it will work.

If it does not work, it means that point clouds used for the network and the point clouds used for searching correspondences are different.

**Comment by AlanConnorChen** [comment]
> The answer is sure! Something should be misconfigured.
> 
> It should be very easily reproduced unless I forgot to update something in the github version.
> 
> Could you please load an examlpe via the datasets/liver_task3.py and check their dimensions?

Thank you for your patience and helpful response!

The issue was caused by the batch size setting. I changed the batch size from 1 to 8 in the original `new_task3_004_002.yaml`. 

The current `collate_fn_descriptor` function in `dataloader.py` is designed for batch size of 1 (I had accidentally commented out the check `assert len(list_data) == 1`), so only the last sample's correspondences, point clouds, and transformations are kept. Meanwhile, `input_points[0]` contains all samples concatenated (which the network processes), causing the dimension mismatch. 

After changing the batch size back to 1, the network trains normally.

<img width="956" height="55" alt="Image" src="https://github.com/user-attachments/assets/893baf50-15d9-47f7-aa49-479f6618fdde" />

If we want `collate_fn_descriptor` to handle batch size > 1, we simply need to concatenate the data from multiple samples.

**Comment by zixinyang9109** [comment]
Interesting and excellent!
